### ISSUE TITLE
On **derive** auto-upgrade BY-* licence to latest publishable version

### ISSUE THREAD

**Issue by pumazi** [Connexions/cnx-authoring#227]
On a derive, authoring should upgrade the BY-* license to the latest version of the license.

For example, if a user derives content that is *BY-CC 3.0*, we should (currently) upgrade to *BY-CC 4.0*.

**Comment by reedstrm** [comment]
Right - legacy has a method 'getDefaultLicense(code='by') that takes an optional code parameter, and returns the currently valid to publish license version with that code. that allows for upgrade-to-the-same-type-only.

**Comment by pumazi** [comment]
This definitely something that has been missed. My previous understanding was that all licenses were upgraded as part of a data upgrade. I'm glad to see that isn't the case.

**Comment by reedstrm** [comment]
Right, we don't change licenses on existing published content. We upgrade at (re)publish time.
